### DISCUSSION TITLE
Have some trouble to configuring retype on a vps with a domain

### DISCUSSION THREAD

**Discussion by ghost** [retypeapp/retype#163] [Q&A]
Hello, so first of all thanks you for your amazing work on retype. I installed a localhost instance of retype and it's works well. 
I also installed it on my vps, but when I run `retype watch`, it tells me to connect to localhost:5000, even after editing the retype.yml with the "url" field ([https://contactit.fr/docs](https://contactit.fr/docs)).

![vivaldi_KPeh5c1dql](https://user-images.githubusercontent.com/61574808/135731030-3d41e673-c8de-4b71-8646-20bdfe509e8a.png)

I've been turning this around for hours, but I still can't connect to my index.md page.
Thanks you in advance.

![WinSCP_EBHNcjUOOF](https://user-images.githubusercontent.com/61574808/135731195-a10a4df3-45ab-412b-9939-5532a66c397a.png)
![WinSCP_jJuNPWuzv6](https://user-images.githubusercontent.com/61574808/135731197-c28d3ea6-b319-4b2c-a4da-2411c80f7a5e.png)

I have an empty .retype folder in my project directory...

![notepad++_nOuXXoEq9B](https://user-images.githubusercontent.com/61574808/135731213-680e82d0-cbf6-49c1-943b-99e2bfe7a456.png)

**Comment by AndreyChechel** [comment]
Hi @febreze21!

Thanks for the kind words about Retype, I'm glad you liked it!

Are you trying to host a website using `retype watch`? It's not really meant to be a production web server, and more considered like a development web server for building a website content. The recommended approach is that a website is first compiled with `retype build` and then the output from the `.retype` folder is hosted with whatever static web server you like.

By default, `retype watch` uses the `in-memory` output storage to produce less IO load on your disk. That's why you were not able to locate the `.retype` folder. This can be changed by setting the [server.watch.mode](https://retype.com/configuration/project/#watchmode) config to `disk`. 

As for the `retype watch` command - the host/port can be configured using the [server](https://retype.com/configuration/project/#server) configuration option - it's probably what you're looking for. By default it's `localhost:5000`.

The [url](https://retype.com/configuration/project/#url) configuration still needs to point to the website root, like you've already configured. So Retype can generate a proper URLs for SEO and the sitemap.

Hope this helps.

**Comment by geoffreymcgill** [comment]
Hi @febreze21 

I'll add a few more comments to thread, likely reiterating some of the above, but maybe it will be helpful too.

Retype does not have to be installed on your server. Retype is only required to build your website, which produces a static .html website that can be deployed onto any web server. 

The actual domain of that website would be configured outside of Retype. For instance, you would need to point your domain name to the website with your DNS settings.

Once you know the location your Retype will be deployed (i.e. https://contactit.fr/docs), then you would set the `url` config with that URL. 

```md
url: https://contactit.fr/docs
```

On your local machine where your Retype project is under development, run the command `retype build` instead of `retype watch`. 

With the `retype build` command, Retype will generate the static website and add everything into a new `.retype` folder. Only the contents of that `.retype` folder would then be deployed (copied, FTP, etc) to your VPS.

During `retype watch`, Retype uses a built in web server and by default all the .html files are stored in memory. 

As mentioned by @AndreyChechel above, you can config Retype to copy all the files to disk during `retype watch` by setting the [`server.watch.mode`](https://retype.com/configuration/project/#watchmode) config. 

```md
server:
  watch:
    mode: disk
```

If others are reading this thread and using GitHub Pages to host their Retype generated website, please see the Retype [Action](https://retype.com/guides/github-actions/) for GitHub Pages.

Hope this helps.

**Reply by ghost** [reply]
Hello and thanks you for your answer.
I builded and ran the project --> All directories appear.
I set a custom host and port:
![notepad++_k7AXowUmYh](https://user-images.githubusercontent.com/61574808/135750760-ef8cda7a-1699-462a-b7c8-3e82ef184ac5.png)
I don't know if there is a way to don't put a port and just use a subdomain instead.
With the port: [https://docs.contactit.fr:5050](https://docs.contactit.fr:5050) is unreachable
Without:  [https://docs.contactit.fr](https://docs.contactit.fr) its shows the aboresence
![vivaldi_VtBRNKQkLN](https://user-images.githubusercontent.com/61574808/135750944-86e2941b-04f0-41c6-8eb0-73ef30f1cdb6.png)

**Comment by geoffreymcgill** [comment]
How you currently have this setup will not work.

The [`server`](https://retype.com/configuration/project/#server) configuration in the retype.yml file is for the internal server, and that server should not be used to host your website. 

The basic steps required to build and host your website on a public accessible web server would be:

1. Build your Retype website on your development machine using `retype build`
2. Copy the generated files from inside the new generated `.retype` folder to your actual live web server
3. DNS and port configuration would be setup on your live web server and DNS provider, not within `retype.yml`. 

Again, any configuration within `retype.yml` is ONLY for local development and testing, not your live website.

**Reply by ghost** [reply]
Oh, okay i see thanks you

**Reply by ghost** [reply]
Its works good in my local machine. I notice that when I transfer the content of the .retype to my web server, I can see the index.html but it can't find the necessary files because of the "/"

![firefox_S4qXngF8cl](https://user-images.githubusercontent.com/61574808/135756557-0e353951-a3ea-4f62-ae25-f91a4829473d.png)

index.html:
![notepad++_41rWxx6LF9](https://user-images.githubusercontent.com/61574808/135756432-2bcdc5e0-6b82-47ce-b121-4dbd0eb6f82b.png)

![WinSCP_kMRfjF1FQs](https://user-images.githubusercontent.com/61574808/135756682-fd3b413e-b105-40b8-97f2-103e77f9cea2.png)

When I remove them all works correctly on the vps.
![firefox_jApPBSxbCP](https://user-images.githubusercontent.com/61574808/135756589-c54149f7-6a97-44bf-9544-b224e848ffc7.png)
Is there a way to fix it please ?

**Comment by geoffreymcgill** [comment]
This seems like an issue with the setup on your VPS. If the URL is `docs.contactit.fr` and the website is setup correctly on the VPS, the virtual rooted path to `/resources` should not be a problem.

As well, please ensure you set the [`url`](https://retype.com/configuration/project/#url) config in your `retype.yml` to `docs.contactit.fr`.

```md
url: docs.contactit.fr
```

Setting the `url` config will not solve this problem, but I do see that you currently have it configured incorrectly for your domain. Should be `url: docs.contactit.fr` not `url: contactit.fr`.

What VPS service are you using? maybe we could find something in their docs to help with the setup.

Hope this helps.

**Reply by ghost** [reply]
docs.contactit.fr point to contactit.fr/docs .I using o2switch.

**Reply by ghost** [reply]
i replaced the url field with docs.contactit.fr and it's works. Thanks you for all. 

I'm volunteer for a French translation of retype, if you need it. Contact me at: pro@contactit.fr I would be happy to contribute to the project.

**Comment by geoffreymcgill** [comment]
Awesome! Happy to hear the website is working correctly now.

Thanks for the translation offer. The functionality is a huge priority for us, although we're still a few releases away from being able to implement.

**Reply by ghost** [reply]
I would like to know if it's possible to copy (with the button) just a section in a code block, per ex
![firefox_I4jGpalBY3](https://user-images.githubusercontent.com/61574808/135773400-3f6b1427-f8b2-485d-839d-5caafad4b61d.png)

**Reply by geoffreymcgill** [reply]
No, sorry. The `copy` button copies the entire contents of the code block.
